smart-contract/assembly/contracts/internals/location.ts

```diff
@@ -0,0 +1,44 @@
+import { bytesToString, stringToBytes } from '@massalabs/as-types';
+import { Storage } from '@massalabs/massa-as-sdk';
+import {
+  fileMetadataKey,
+  fileMetadataLocationKey,
+} from './storageKeys/metadataKeys';
+import { FILE_METADATA_LOCATION_TAG } from './storageKeys/tags';
+
+/**
+ * Adds a new file location to the list of file locations.
+ * @param location - The file location to be added.
+ * @param hashLocation - The hash of the file location.
+ */
+export function _pushFileLocation(
+  location: string,
+  hashLocation: StaticArray<u8>,
+): void {
+  Storage.set(fileMetadataLocationKey(hashLocation), stringToBytes(location));
```

Comment:
This should not be a separate datastore prefix, it should be one metadata entry among others for the file. No need to define a special way to handle it, simply reuse the metadata related functions

Reply:
We could state the rule:
* if the metadata key starts with "http_header:" then it is a http header
* use the key "location" for the location
* also support other keys as defined in https://forum.massa.community/t/community-feedback-proposal-for-improving-website-storage-on-deweb/82/15#p-319-metadata-meaning-3 but this is transparent to the smart contract and only handled by the client